src/client/components/editor/index.js

```diff
@@ -343,9 +346,9 @@ class Editor extends React.Component {
   }
 
   resetMenuState(){
-    this.setData({
-      drawMode: false,
-      taskListMode: false
+    return Promise.all([this.toggleTaskListMode(false),  this.toggleDrawMode(false)]).then( () => {
+      // wait for menu animations to complete
+      return new Promise( resolve => setTimeout(resolve, 250));
```

Comment:
There's a Promise.delay(250) in bluebird you can use that does the exact same thing.  May be useful in future, but not absolutely necessary to change now